test/onair/data_handling/test_redis_adapter.py

```diff
@@ -684,3 +673,62 @@ def test_redis_adapter_DataSource_process_data_file_does_nothing():
 
     # Assert
     assert result == expected_result
+
+
+def test_redis_adapter_DataSource_parse_meta_data_file_redis_in_keys_subscriptions_exist_and_adds_redis_to_server_configs(mocker):
+    # Arrange
+    cut = DataSource.__new__(DataSource)
+    arg_configFile = MagicMock()
+    arg_ss_breakdown = MagicMock()
+
+    fake_server_configs = [{"subscriptions": ["state_0", "state_1"]}, {"subscriptions": ["state_2", "state_3"]}]
+    cut.server_configs = MagicMock()
+
+    expected_extracted_configs = MagicMock()
+    fake_meta = {'fake_other_stuff': MagicMock(),
+                  'redis': fake_server_configs,
+                  'order': MagicMock()}
+
+    mocker.patch(redis_adapter.__name__ + '.extract_meta_data_handle_ss_breakdown', return_value=expected_extracted_configs)
+    mocker.patch(redis_adapter.__name__ + '.parseJson', return_value=fake_meta)
+
+    # Act
+    result = cut.parse_meta_data_file(arg_configFile, arg_ss_breakdown, )
+
+    # Assert
+    assert redis_adapter.extract_meta_data_handle_ss_breakdown.call_count == 1
+    assert redis_adapter.extract_meta_data_handle_ss_breakdown.call_args_list[0].args == (arg_configFile, arg_ss_breakdown)
+    assert redis_adapter.parseJson.call_count == 1
+    assert redis_adapter.parseJson.call_args_list[0].args == (arg_configFile, )
+    assert result == expected_extracted_configs
+    assert cut.server_configs == fake_meta['redis']
+
+def test_redis_adapter_DataSource_parse_meta_data_file_redis_in_keys_subscriptions_do_not_exist(mocker):
+    # Arrange
+    cut = DataSource.__new__(DataSource)
+    arg_configFile = MagicMock()
+    arg_ss_breakdown = MagicMock()
+
+    fake_server_configs = [{"address": 1}]
+    cut.server_configs = MagicMock()
+
+    expected_extracted_configs = MagicMock()
+    fake_meta = {'fake_other_stuff': MagicMock(),
+                  'redis': fake_server_configs,
+                  'order': MagicMock()}
+
+    mocker.patch(redis_adapter.__name__ + '.extract_meta_data_handle_ss_breakdown', return_value=expected_extracted_configs)
+    mocker.patch(redis_adapter.__name__ + '.parseJson', return_value=fake_meta)
+
+    # Act
+    with pytest.raises(ConfigKeyError) as e_info:
+        cut.parse_meta_data_file(arg_configFile, arg_ss_breakdown, )
+
+
+    # Assert
+    assert redis_adapter.extract_meta_data_handle_ss_breakdown.call_count == 1
+    assert redis_adapter.extract_meta_data_handle_ss_breakdown.call_args_list[0].args == (arg_configFile, arg_ss_breakdown)
+    assert redis_adapter.parseJson.call_count == 1
+    assert redis_adapter.parseJson.call_args_list[0].args == (arg_configFile, )
+    assert cut.server_configs == []
+    assert e_info.match(f'Config file: \'{arg_configFile}\' missing required key \'subscriptions\' from 0 in key \'redis\'')
```

Comment:
Newline at the end of the file, please. Did this go through pylint/black?

Reply:
It did not yet as it was submitted prior to #139 approval.

Reply:
Updated code, conforming to pylint and black.